Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -1864,21 +1864,45 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
     ///   - bundle: The bundle containing the articles.
     private func synthesizeArticleOnlyRootPage(articles: inout [DocumentationContext.SemanticResult<Article>], bundle: DocumentationBundle) {
         let title = bundle.displayName
-        let metadataDirectiveMarkup = BlockDirective(name: "Metadata", children: [
-            BlockDirective(name: "TechnologyRoot", children: [])
-        ])
-        let metadata = Metadata(from: metadataDirectiveMarkup, for: bundle, in: self)
+        
+        // An inner helper function to register a new root node from an article
+        func registerAsNewRootNode(_ articleResult: SemanticResult<Article>) {
+            uncuratedArticles.removeValue(forKey: articleResult.topicGraphNode.reference)
+            let title = articleResult.source.deletingPathExtension().lastPathComponent
+            // Create a new root-looking reference
+            let reference = ResolvedTopicReference(
+                bundleIdentifier: bundle.identifier,
+                path: NodeURLGenerator.Path.documentation(path: title).stringValue,
+                sourceLanguages: [DocumentationContext.defaultLanguage(in: [])]
+            )
+            // Add the technology root to the article's metadata
+            let metadataMarkup: BlockDirective
+            if let markup = articleResult.value.metadata?.originalMarkup as? BlockDirective {
+                metadataMarkup = markup.withUncheckedChildren(
+                    markup.children + [BlockDirective(name: "TechnologyRoot", children: [])]
```

Comment:
How do we know the existing article doesn't already have a `@TechnologyRoot` directive? I suppose because this entire code path under `synthesizeArticleOnlyRootPage` wouldn't occur if it did. But we might want to add an assertion here that the existing article doesn't already have `@TechnologyRoot` so we can be sure we aren't adding a duplicate directive.

Reply:
Correct. The context wouldn't call `synthesizeArticleOnlyRootPage` (which is a private method) if it already had an explicitly authored root page or module page. 

Still, I added a debug assertion here to help catch potentially mistakes in the future.